redis/docs/token_authentication.md

```diff
@@ -0,0 +1,225 @@
+# Token-Based Authentication with Azure Entra ID
```

Comment:
instead of adding this as a separate document in a new folder, let's try to integrate this into the library's rustdoc, in a way that will use our ability to check that the code actually compiles. Please move this to the `entra_id` module, and check the generated rust docs to see that it looks like you expect the doc to look, and that each named type is linked to the type's documentation.